File: ValidPalindrome/ValidPalindrome.c
bool validChar(char c);
bool isPalindrome(char* s) {
    int front, back;
    front = 0;
    back = strlen(s) - 1;
    while (front < back){
        if(!validChar(s[front]))
            front ++;
        else if(!validChar(s[back]))
            back --;
        else if(tolower(s[front]) != tolower(s[back]))
            return false;
        else{
            front ++;
            back --;
        }
        
    }
    return true;
    
}
bool validChar(char c){
    return (c >= 'a' && c <= 'z') ||(c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9') ;
}

File: ValidPalindrome/ValidPalindrome.java
public class ValidPalindrome {
    public boolean isPalindrome(String s) {
        int front, back;
        front = 0; 
        back = s.length() - 1;
        while(front < back){
            if(!validChar(s.charAt(front)))
                front ++;
            if(!validChar(s.charAt(back)))
                back --;
            if (validChar(s.charAt(back)) &&validChar(s.charAt(front))){
                if(Character.toLowerCase(s.charAt(back)) != Character.toLowerCase(s.charAt(front)))// case sensitive
                    return false;
                front++;
                back --;
            }
        }
        return true;
    }
    
    private boolean validChar (char c){
        return Character.isDigit(c)|| Character.isLetter(c);
    }
}